lms/djangoapps/teams/static/teams/js/views/edit_team.js

```diff
@@ -0,0 +1,168 @@
+                   this.optionalDescriptionField = new FieldViews.ReadonlyFieldView({
+                       model: new TeamModel(),
+                       title: gettext("Optional Characteristics"),
+                       valueAttribute: 'optional_description',
+                       helpMessage: gettext("Help other learners decide whether to join your team by specifying some characteristics for your team. Choose carefully, because fewer people might be interested in joining your team if it seems too restrictive. You cannot change these characteristics after you create the team.")
+                   });
+
+                   this.teamLanguageField = new FieldViews.DropdownFieldView({
+                       model: new TeamModel(),
+                       title: gettext("Language"),
+                       valueAttribute: 'language',
+                       required: false,
+                       showMessages: false,
+                       bindEvents: false,
+                       titleIconName: 'fa-comment-o',
+                       options: this.languages,
+                       helpMessage: gettext("The language that team members primarily use to communicate with each other.")
+                   });
+
+                   this.teamCountryField = new FieldViews.DropdownFieldView({
+                       model: new TeamModel(),
+                       title: gettext('Country'),
+                       valueAttribute: 'country',
+                       required: false,
+                       showMessages: false,
+                       bindEvents: false,
+                       titleIconName: 'fa-globe',
+                       options: this.countries,
+                       helpMessage: gettext("The country that team members primarily identify with.")
+                   });
+               },
+
+               render: function() {
+                   this.$el.html(_.template(edit_team_template)({}));
+                   this.set(this.teamNameField, '.team-required-fields');
+                   this.set(this.teamDescriptionField, '.team-required-fields');
+                   this.set(this.optionalDescriptionField, '.team-optional-fields');
+                   this.set(this.teamLanguageField, '.team-optional-fields');
+                   this.set(this.teamCountryField, '.team-optional-fields');
+                   return this;
+               },
+
+               set: function(view, selector) {
+                   this.$el.find(selector).append(view.$el);
+                   view.render();
+               },
+
+               createTeam: function () {
+                   var teamName = this.teamNameField.fieldValue();
+                   var teamDescription = this.teamDescriptionField.fieldValue();
+                   var teamLanguage = this.teamLanguageField.fieldValue();
+                   var teamCountry = this.teamCountryField.fieldValue();
+
+                   var validation = this.validateTeamData(teamName, teamDescription);
```

Comment:
Why not create the `data` dictionary first and pass it to `validataTeamData`. This way, if we need to add more fields or other checks, this call will not need to change.